### DISCUSSION TITLE
Extension host did not start in 10 seconds

### DISCUSSION THREAD

**Discussion by umar-shabbir** [microsoft/vscode-discussions#826] [Extension Development QnA] [answered]
I often get this error when starting the extension development host window by pressing F5 which is the usual way to debug an extension as described in the VS code documentation. Following is the full error message:
**Extension host did not start in 10 seconds, it might be stopped on the first line and needs a debugger to continue.**

I have tried following which did not solve this problem:
- Adding timeout key in the launch.json with greater value like 20000ms. Still the error says 10 seconds.
- Allowing vs code through firewall
- Disconnecting any VPN's
- Removing all my custom code and trying it with the Hello World sample from VS code samples

I am unable to further figure out the root cause of this error.

Here is some output related to this error as seen in extension development host window. Its something related to ports but I have not made any such changes like specifying the ports. Moreover, it looks like some internal VS code error.

`2023-09-18 18:37:16.358 [error] [LocalProcessExtensionHost]: Extension host did not start in 10 seconds (debugBrk: true)
2023-09-18 18:37:16.705 [info] [perf] Render performance baseline is 356ms
2023-09-18 18:37:44.837 [error] [uncaught exception in main]: Port should be >= 0 and < 65536. Received type number (65964).: RangeError [ERR_SOCKET_BAD_PORT]: Port should be >= 0 and < 65536. Received type number (65964).
    at new NodeError (node:internal/errors:399:5)
    at validatePort (node:internal/validators:390:11)
    at lookupAndConnect (node:net:1227:5)
    at Socket.connect (node:net:1193:5)
    at s (C:\Users\m3nbc8\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\code\electron-main\main.js:19:93503)
    at Socket.<anonymous> (C:\Users\m3nbc8\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\code\electron-main\main.js:19:93392)
    at Object.onceWrapper (node:events:627:28)
    at Socket.emit (node:events:525:35)
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1479:10)`

**Comment by theReynald** [comment, marked as answer]
Hi @umar-shabbir,

The error message you're encountering suggests that the extension host is taking longer than expected to start, and it mentions an issue with a port number that exceeds the valid range. This can happen for various reasons, and it's not always straightforward to diagnose. However, there are a few common steps you can take to troubleshoot and resolve this issue:

1. **Check for Extension Conflicts:**
   Sometimes, conflicts between extensions can lead to issues with the extension host startup. Try disabling other extensions one by one to see if any specific extension is causing the problem. This will help identify if the issue is related to a particular extension.

2. **Check for Large Workspace/Project:**
   If you are working on a large workspace or project, it can take a significant amount of time for the extension host to start. You might want to try opening a smaller or empty workspace to see if the issue persists.

3. **Update VS Code and Extensions:**
   Ensure that you are using the latest version of Visual Studio Code and all installed extensions are up to date. Outdated extensions can sometimes lead to compatibility issues.

4. **Check for Workspace Configuration Issues:**
   Verify your workspace's configuration files (e.g., `settings.json`, `launch.json`) for any errors or inconsistencies. An improperly configured workspace can cause issues with the extension host startup.

5. **Increase Extension Host Timeout:**
   You mentioned that you tried increasing the timeout in your `launch.json` file. Make sure you've modified it correctly. Here's an example of how it should look:

   ```json
   {
       "version": "0.2.0",
       "configurations": [
           {
               "name": "Extension Host",
               "type": "extensionHost",
               "request": "launch",
               "runtimeExecutable": "${execPath}",
               "args": [
                   "--extensionDevelopmentPath=${workspaceFolder}"
               ],
               "timeout": 20000 // Increase the timeout to 20 seconds
           }
       ]
   }
   ```

6. **Check for Firewall/Antivirus Interference:**
   Firewalls or antivirus software can sometimes block certain processes or ports used by Visual Studio Code. Make sure your firewall or antivirus is not interfering with VS Code's operation.

7. **Try Running VS Code as Administrator:**
   Running Visual Studio Code with administrative privileges can sometimes resolve permission-related issues.

8. **Reinstall Visual Studio Code:**
   As a last resort, consider uninstalling and reinstalling Visual Studio Code. This can help resolve any corrupted installation files or configurations.

9. **Review System Resources:**
   Check your system's resource utilization (CPU, memory, etc.) while starting VS Code. Insufficient system resources can slow down the startup process.

Hope this helps.

**Reply by lppedd** [reply]
Is there a way to set the timeout when starting the extension through CLI?
